scipy/cluster/vq.py

```diff
@@ -137,8 +137,8 @@ def whiten(obs, check_finite=True):
     obs = _asarray(obs, check_finite=check_finite, xp=xp)
     std_dev = xp.std(obs, axis=0)
     zero_std_mask = std_dev == 0
-    if xp.any(zero_std_mask):
-        std_dev[zero_std_mask] = 1.0
+    std_dev = xpx.at(std_dev, zero_std_mask).set(1.0)
+    if check_finite and xp.any(zero_std_mask):
```

Comment:
This fails on jax.jit with the default value.
https://github.com/data-apis/array-api-compat/issues/225 is a blocker for a more robust fix.